actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -164,79 +164,72 @@ pub fn call<BS: Blockstore, RT: Runtime<BS>>(
             let dst_addr: EthAddress = dst.try_into()?;
             let dst_addr: Address = dst_addr.try_into()?;
 
-            // Special casing for embryo/non-existent actors: we just do a SEND (method 0)
+            // Special casing for account/embryo/non-existent actors: we just do a SEND (method 0)
             // which allows us to transfer funds (and create embryos)
-            let is_embryonic = if dst_addr.protocol() == AddressProtocol::ID {
-                // sanity check: this shouldn't be an ID address, as you can't predict
-                // what actor is gonna sit there.
-                false
-            } else if let Some(actor_id) = system.rt.resolve_address(&dst_addr) {
-                if let Some(cid) = system.rt.get_actor_code_cid(&actor_id) {
-                    system.rt.resolve_builtin_actor_type(&cid) == Some(Type::Embryo)
-                } else {
-                    true
-                }
-            } else {
-                true
-            };
-
-            let call_result = if is_embryonic {
-                system.send(
-                    &dst_addr,
-                    0,
-                    // we still send the input, even thought it will be ignored, for debugging
-                    // purposes
-                    RawBytes::serialize(BytesSer(input_data))?,
-                    TokenAmount::from(&value),
-                )
-            } else {
-                match kind {
-                    CallKind::Call => system.send(
+            let target_actor_code = system
+                .rt
+                .resolve_address(&dst_addr)
+                .and_then(|actor_id| system.rt.get_actor_code_cid(&actor_id))
+                .and_then(|cid| system.rt.resolve_builtin_actor_type(&cid));
+
+            let call_result =
+                if matches!(target_actor_code, Some(Type::Embryo | Type::Account) | None) {
```

Comment:
wait, you dropped the non existent id address check, was that on purpose or oversight?

Reply:
No, but it doesn't really matter either way.

- The old code would just do a normal call, which will fail.
- This code will do a method 0 send, which will fail.

Basically, I was trying to simplify a bit (and handle the remaining edge cases).